graph/history.go

```diff
@@ -38,8 +37,7 @@ func (graph *Graph) depth(img *image.Image) (int, error) {
 
 	for parent != nil {
 		count++
-		parent, err = graph.GetParent(parent)
-		if err != nil {
+		if parent, err = graph.GetParent(parent); err != nil {
```

Comment:
you could even remove `err error` declaration above https://github.com/docker/docker/pull/16103/files#diff-834986d10d3853e020577ea23e648c83R35 and do `if parent, err :=` here (with colon)